packages/babel-plugin-mui-css/src/index.ts

```diff
@@ -0,0 +1,210 @@
+      }
+      if (a.filepath in indexByPath) {
+        return -1;
+      }
+      if (b.filepath in indexByPath) {
+        return +1;
+      }
+      return a.filepath.localeCompare(b.filepath);
+    });
+
+    let cssContent = '';
+
+    cssContent = state.cssRules.map((r) => r.content).join('\n');
+
+    if (state.cssMinify) {
+      const { code: cssMinified } = lightning({
+        filename: 'index.css',
+        code: Buffer.from(cssContent),
+        minify: true,
+      });
+      cssContent = cssMinified as any;
+    }
+
+    const outputPath = joinPath(process.env.MUI_CSS_OUTPUT_DIR ?? '', state.cssOutput);
+
+    mkdirp.sync(dirname(outputPath));
+    writeFileSync(outputPath, cssContent);
+  });
+
+  return {
+    name: PLUGIN_NAME,
+
+    manipulateOptions(options: any) {
+      const plugin = findSelf(options.plugins);
+      let content: string | undefined;
+      try {
+        content = readFileSync('mui-css.config.json').toString();
+      } catch (error) {
+        // ignore
+      }
+      if (content) {
+        const newOptions = JSON.parse(content);
+        plugin.options = { ...plugin.options, ...newOptions };
+      }
+      return options;
+    },
+
+    pre(file) {
+      const opts = findSelf(file.opts.plugins).options;
+
+      if (!state.cssVariables) {
+        state.cssVariables = getCSSVariablesCode(opts?.cssVariables);
+      }
+
+      state.cssMinify = opts.cssMinify ?? true;
+      state.cssOutput = opts.cssOutput ?? 'index.css';
+    },
+
+    async post(file) {
+      if (!state.indexByPath) {
+        const configPath = packageDirectorySync({ cwd: dirname(file.opts.filename) });
+        const mainPath = JSON.parse(
+          readFileSync(joinPath(configPath, 'package.json')).toString(),
+        ).main;
+        state.indexByPath = await createImportMap(joinPath(configPath, mainPath));
+      }
+    },
+
+    visitor: {
+      // const styles = css('prefix', {
+      //   class1: { ... },
+      // });
+      CallExpression(path, { file }) {
+        const {
+          callee: { name: calleeName },
+          arguments: args,
+        } = path.node;
+
+        if (calleeName !== 'css') {
+          return;
+        }
+        if (!t.isStringLiteral(args[0])) {
+          throw new Error(`Invalid CSS prefix: ${formatLocation(file, path.node)}`);
+        }
+        if (!t.isObjectExpression(args[1])) {
+          throw new Error(`Invalid CSS styles: ${formatLocation(file, path.node)}`);
+        }
+
+        const [prefixNode, classesNode] = args;
+
+        const prefix = prefixNode.extra.rawValue;
+
+        /* eslint-disable no-underscore-dangle */
+        const context = {
+          __source: file.code.slice(classesNode.start, classesNode.end),
+          __result: null,
+        };
+        vm.createContext(context);
+        vm.runInContext(
+          `
+          ${state.cssVariables};
+          __result = eval('(' + __source + ')')
```

Comment:
Isn't the `eval('(' + __source + ')')` redundant inside of `vm.runInContext`? Couldn't you just do


```tsx
       const source = file.code.slice(classesNode.start, classesNode.end)
       vm.runInContext(
          `
          ${state.cssVariables};
          __result = (${source});
        `,
          context,
        );
```